ha/resource/dynamic_fid_service_ra.py

```diff
@@ -0,0 +1,266 @@
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+"""
+ ****************************************************************************
+ Description:       Generic resource agent used to map fid to cloneids.
+ ****************************************************************************
+"""
+
+import os
+import sys
+import time
+import pathlib
+
+from cortx.utils.log import Log
+
+sys.path.append(os.path.join(os.path.dirname(pathlib.Path(__file__)), '..', '..'))
+from ha.core.config.config_manager import ConfigManager
+from ha.resource.resource_agent import CortxServiceRA
+from ha.execute import SimpleCommand
+from ha.plugin.hare.hare import Hare
+from ha import const
+
+class FidManager:
+    """
+    Manage Fid for different services.
+    """
+
+    @staticmethod
+    def get_fid(service: str, node_id: str, instance_id: int) -> str:
+        """
+        Get Fid from hare mapping file for given services.
+
+        Args:
+            service ([str]): Service name.
+            node_id ([str]): Node name for fid instance.
+            instance_id ([int]): Instance id for service.
+
+        Returns:
+            str: Return fid of given service.
+        """
+        return Hare.get_fid(service, node_id, instance_id)
+
+class DynamicFidServiceRA(CortxServiceRA):
+    """
+    This class is used to provide wrapper around systemd resource agent.
+    This class manage fid mapping to (serviceName, NodeId, InstanceId).
+    Resource name: (Used to get instance)
+        like motr-ios-1, s3server-1, ...
+        Here, resource name is used to get instance.
+    Service name: (Used for systemd service)
+        like s3server, m0d, ...
+    Fid service name: (Used for service name in mapping)
+        like s3service, ios, confd
+    """
+    def __init__(self):
+        """
+        Initialize DynamicFidServiceRA class.
+        """
+        super(DynamicFidServiceRA, self).__init__()
+        self._execute = SimpleCommand()
+        self._status_list: list = ["failed", "active", "unknown"]
+
+    def _get_systemd_service(self) -> str:
+        """
+        Get Service name.
+
+        Returns:
+            str: Service name with fid mapping like service@fid
+        """
+        res_param = self.get_env()
+        service: str = res_param["OCF_RESKEY_service"]
+        fid_service_name: str = res_param["OCF_RESKEY_fid_service_name"]
+        local_node: str = res_param["OCF_RESKEY_CRM_meta_on_node"]
+        resource: str = res_param["OCF_RESOURCE_INSTANCE"]
+        instance_id: int = int(resource.split('-')[-1])
+        fid = FidManager.get_fid(fid_service_name, local_node, instance_id)
```

Comment:
Exception should be caught as well and to be handled in same way as fid = None.

Reply:
handled exception in FidManager we will either get fid or None here